core/src/main/scala/caliban/GraphQLRequest.scala

```diff
@@ -0,0 +1,25 @@
+package caliban
+
+import caliban.interop.circe._
+
+import scala.language.higherKinds
+
+/**
+ * Represents a GraphQL request, containing a query, an operation name and a map of variables.
+ */
+case class GraphQLRequest(
+  query: String,
+  operationName: Option[String],
+  variables: Option[Map[String, InputValue]]
+)
+
+object GraphQLRequest {
+  implicit def circeDecoder[F[_]: IsCirceDecoder]: F[GraphQLRequest] =
+    GraphQLRequestCirce.graphQLRequestDecoder.asInstanceOf[F[GraphQLRequest]]
+}
+
+private object GraphQLRequestCirce {
+  import io.circe._
+  import io.circe.derivation._
```

Comment:
Since we are using circe-derivation only for this single (and simple) type, how about deriving it manually and only depend on circe-core?
Because not everyone uses circe-derivation, some people use circe-generic, others circe-magnolia. That way we would keep the dependency minimal.

Reply:
I think codecs created by circe-derivation macros depend on the circe core only, so people won't need circe-derivation as a dependency. Although, to be on the safe side I can indeed implement it manually.

Reply:
Oh yeah, you must be right! All good then.